jdaviz/configs/specviz/helper.py

```diff
@@ -39,17 +39,31 @@ def get_spectra(self, data_label=None, apply_slider_redshift="Warn"):
         if not apply_slider_redshift:
             return spectra
         else:
-            if data_label is None:
-                for key in spectra.keys():
-                    spectra[key].redshift = self._redshift
-            else:
-                spectra.redshift = self._redshift
+            output_spectra = {}
+            # We need to create new Spectrum1D outputs with the redshifts set
+            if data_label is not None:
+                spectra = {data_label: spectra}
+            for key in spectra.keys():
+                flux = spectra[key].flux
+                # This is a hack around inability to input separate redshift with
+                # a SpectralAxis instance in Spectrum1D
+                spaxis = spectra[key].spectral_axis.value * spectra[key].spectral_axis.unit
+                mask = spectra[key].mask
+                uncertainty = spectra[key].uncertainty
+                print(self._redshift)
```

Comment:
I would suggest clarifying the print statement (i.e. "Redshift: ") or removing it altogether.

Reply:
Good catch, this was a leftover from debugging.